pkg/model/hetznermodel/network.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hetznermodel
+
+import (
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/hetzner"
+	"k8s.io/kops/upup/pkg/fi/cloudup/hetznertasks"
+)
+
+// NetworkModelBuilder configures network objects
+type NetworkModelBuilder struct {
+	*HetznerModelContext
+	Lifecycle fi.Lifecycle
+}
+
+var _ fi.ModelBuilder = &NetworkModelBuilder{}
+
+func (b *NetworkModelBuilder) Build(c *fi.ModelBuilderContext) error {
+	network := &hetznertasks.Network{
+		Name:      fi.String(b.ClusterName()),
+		Lifecycle: b.Lifecycle,
+		IPRange:   fi.String(b.Cluster.Spec.NetworkCIDR),
+		Subnets: []string{
+			b.Cluster.Spec.NetworkCIDR,
```

Comment:
In future, maybe this should be subnets.  But I agree no need today, and that the sensible default is that the subnet CIDR should == the network CIDR.  If we support e.g. vswitch in future, users might have to specify subnet CIDRs to use that.